src/util.js

```diff
@@ -107,3 +107,55 @@ export function format_memory_and_limit(usage, limit) {
         return _("");
     }
 }
+
+export function updateContainers() {
+    let newContainers = {};
+    let newContainersStats = {};
+    return new Promise((resolve, reject) => {
+        varlinkCall(PODMAN, "io.podman.ListContainers")
+                .then(reply => {
+                    let newContainersMeta = reply.containers;
+                    let inspectRet = newContainersMeta.map((container) => {
+                        return new Promise((resolve, reject) => {
+                            varlinkCall(PODMAN, "io.podman.InspectContainer", {name: container.id})
+                                    .then(reply => resolve(JSON.parse(reply.container)))
+                                    .catch(ex => reject(new Error("Failed to do InspectContainer call:", ex, JSON.stringify(ex))));
+                        });
+                    });
+
+                    Promise.all(inspectRet)
+                            .then((inspectRet) => {
+                                inspectRet.map((inspectRetEle) => {
+                                    newContainers[inspectRetEle.ID] = inspectRetEle;
+                                });
+                                resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                            })
+                            .catch(ex => console.error("Failed to do InspectContainer call:", ex, JSON.stringify(ex)));
+
+                    let containerStatsRet = newContainersMeta.filter(ele => ele.status === "running")
+                            .map((container) => {
+                                return new Promise((resolve, reject) => {
+                                    varlinkCall(PODMAN, "io.podman.GetContainerStats", {name: container.id})
+                                            .then(reply => resolve({ctrId: container.id, ctrStats:reply.container}))
+                                            .catch(ex => {
+                                                console.error("Failed to do GetContainerStats call:", ex, JSON.stringify(ex));
+                                                reject(new Error("Failed to do GetContainerStats call:", ex, JSON.stringify(ex)));
+                                            });
+                                });
+                            });
+
+                    Promise.all(containerStatsRet)
+                            .then((containerStatsRet) => {
+                                containerStatsRet.map((containerStatsRetEle) => {
+                                    newContainersStats[containerStatsRetEle.ctrId] = containerStatsRetEle.ctrStats;
+                                });
+                                resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                            })
+                            .catch(ex => console.error("Failed to do GetContainerStats call:", ex, JSON.stringify(ex)));
```

Comment:
Same here, this needs to reject the top-level promise and propagate the exception.